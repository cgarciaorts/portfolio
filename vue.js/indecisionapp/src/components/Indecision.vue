<template>
<div>
    <img :src="imagen" alt="fondo">

    <div class="indecision-container">
        <input v-model="pregunta" type="text" placeholder="Hazme una pregunta">
        <p>Recuerda terminar con un signo de interrogacion (?)</p>
        <div>
            <h2> Hola mundo </h2>
            <h1> {{respuesta}} </h1>
            <h1> cosa 1 </h1>
        </div>
    </div>
</div>

</template>
<script>
export default {
    name: 'Indecision',
    data(){
        return{
            pregunta: 'Soy precioso?',
            respuesta: null,
            imagen: null
        }
    },
    methods: {
        async obtenerRespuesta(){
            this.respuesta = 'Pensando...'
            const {answer, image} = await fetch("https://yesno.wtf/api")
            .then(respuesta => respuesta.json() )
            this.respuesta = answer
            this.imagen = image
        }
    },
    watch: {
        pregunta(value, oldValue){
            if (value.includes('?')) 
            return this.obtenerRespuesta()
        }
    }

}
</script>

<style lang="sass">
    *
        margin: 0
        padding: 0
        box-sizing: border-box
        transition: .5s
    
    img
        height: 100vh
        max-width: 100%
        max-height: 100%
        position: fixed
        width: 100vw

    .indecision-container
        position: relative
        z-index: 99
        display: flex
        justify-content: center
        align-items: center
        flex-direction: column
        background: rgba(0, 0, 0, .3)
        height: 100vh

    input
        border: none
        padding: 1rem
        width: 50%
        border-radius: 15px
        font-size: 20px
        color: #565656
        outline: none
    
    p
        margin: 2%
        color: red
        font-size: 25px
        font-weight: bold
            
</style>