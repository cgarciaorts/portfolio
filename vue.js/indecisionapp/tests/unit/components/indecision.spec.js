import Indecision from '@/components/Indecision'
import { shallowMount } from '@vue/test-utils'

describe ('Indecision Component', () => {
    test('Debe averiguar que existe un elemento hmtl',() =>{
        const componente = shallowMount(Indecision)
        expect(componente.find('h1').exists()).toBeTruthy()
    })    
    test('Debe averiguar que h2 tiene el valor de hola mundo',() =>{
        const componente = shallowMount(Indecision)
        const valorh2 = componente.find('h2').text()//busca en mi componente indecision si hay un h2
        expect(valorh2).toBe("Hola mundo")//se espera que el valor sea hola mundo, si es asi pasarÃ¡ la prueba
    })
    test('Debe mostrar los h1',() =>{
        const componente = shallowMount(Indecision)
        const h1etiquetas = componente.findAll(h1)
        console.log(h1etiquetas)
        expect(h1etiquetas[1].text()).toBe("cosa 1")
    })    

    // test ('Debe hacer match con el snapshot', () => {
    //     const estructura = shallowMount(Indecision) //aqui hace la captura
    //     expect(estructura.html()).toMatchSnapshot() //aqui compara la captura con la que le pasan
    
    // })
})

// describe ('Indecision Component', () => {
//     test ('Debe tener el valor de la api', () => {
//         const estructura = shallowMount(Indecision) //aqui hace la captura
//         const h2 = estructura.find('h2') //aqui hace la captura
//         expect(h2.text()).toBe('Hola mundo')
//     })
// })