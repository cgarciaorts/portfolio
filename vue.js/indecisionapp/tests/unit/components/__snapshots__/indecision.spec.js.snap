// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indecision Component Debe hacer match con el snapshot 1`] = `
<div><img alt="fondo">
  <div class="indecision-container"><input type="text" placeholder="Hazme una pregunta">
    <p>Recuerda terminar con un signo de interrogacion (?)</p>
    <div>
      <h2> HOLA MUNDO </h2>
      <h1></h1>
    </div>
  </div>
</div>
`;
